runtime/backtrace.c

```diff
@@ -30,7 +30,7 @@
 
 void caml_init_backtrace(void)
 {
-  return;
+  caml_register_generational_global_root(&Caml_state->backtrace_last_exn);
```

Comment:
I don't see anywhere in the code where `caml_init_backtrace` is ever called (which is unfortunate as it probably shouldn't be there).

Reply:
Yes, it was not called anywhere, so `Caml_state->backtrace_last_exn` was not being registered as a root. Thanks for finding this! I have attempted to fix this now, `caml_init_backtrace` is called during startup.